tests/on-device/lib.bash

```diff
@@ -11,11 +11,14 @@ setup() {
       fi
     done < <(nix-channel --list)
     echo "parsing detected channel url: $CHANNEL_URL"
-    [[ "$CHANNEL_URL" =~ https://github.com/(.+)/(.+)/archive/(.+)\.tar\.gz ]]
-    REPO_USER=${BASH_REMATCH[1]}
-    REPO_NAME=${BASH_REMATCH[2]}
-    REPO_BRANCH=${BASH_REMATCH[3]}
-    FLAKE_URL=github:$REPO_USER/$REPO_NAME/$REPO_BRANCH
+    if [[ "$CHANNEL_URL" =~ https://github.com/(.+)/(.+)/archive/(.+)\.tar\.gz ]]; then
+      REPO_USER=${BASH_REMATCH[1]}
+      REPO_NAME=${BASH_REMATCH[2]}
+      REPO_BRANCH=${BASH_REMATCH[3]}
+      FLAKE_URL=github:$REPO_USER/$REPO_NAME/$REPO_BRANCH
+    elif [[ "$CHANNEL_URL" == file:///n-o-d/archive.tar.gz ]]; then
+      FLAKE_URL=/n-o-d/unpacked
```

Comment:
Does this directory work as a flake without a git repo in it?

Reply:
Yes, flake doesn't have to be an initialized repo, it's just the behaviour changes significantly when it is.

Reply:
That is good to know! :)